scripts/ghidra/PatchestryDecompileFunctions.java

```diff
@@ -891,14 +912,31 @@ private String findNullTerminatedString(Address address, Pointer pointer) throws
 		}
 
 		private Data getDataReferencedAsConstant(Varnode node) throws Exception {
+			// check if node is null
+			if (node == null) {
+				return null;
+			}
+
+			 // Only process constant nodes that aren't nulls (address 0 in constant space)
 			if (!node.isConstant() || node.getAddress().equals(constant_space.getAddress(0))) {
 				return null;
 			}
+
 			// Ghidra sometime fail to resolve references to Data and show it as const. 
 			// Check if it is referencing Data as constant from `ram` addresspace.
-			Address ram_address = convertAddressToRamSpace(node.getAddress());
-			Data data = getDataAt(ram_address);
-			return data;
+			try {
+				// Convert the constant value to a potential RAM address
+				Address ram_address = convertAddressToRamSpace(node.getAddress());
+				if (ram_address == null) {
+					return null;
+				}
+				return getDataAt(ram_address);
+
+			} catch (AddressOutOfBoundsException e) {
```

Comment:
Same as above - I would really prefer propagating exceptions rather than swallowing them if you don't mind please!

Reply:
We don't need to catch here. removed it